Tools/ApplicationDebugger/jacobi/src/jacobi-fixed.cpp

```diff
@@ -0,0 +1,119 @@
+//==============================================================
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+// The program solves the linear equation Ax=b, where matrix A is a
+// n x n sparse matrix with diagonals [1 1 4 1 1],
+// vector b is set such that the solution is [1 1 ... 1]^T.
+// The linear system is solved via Jacobi iteration.
+
+// Each call of the kernel computes one iteration of the Jacobi method.
+
+#include <CL/sycl.hpp>
+#include <iostream>
+// Location of file: <oneapi-root>/dev-utilities/<version>/include
+#include "dpc_common.hpp"
+#include "selector.hpp"
+
+using namespace std;
+using namespace sycl;
+
+int main(int argc, char *argv[]) {
+  constexpr size_t n = 64;
+  int b[n];
+  float x_k[n];
+  float x_k1[n];
+
+  // Initialize the input.
+  for (int i = 0; i < n; i++) {
+    b[i] = 8;
+    x_k[i] = 0;
+  }
+
+  b[0] = 6;
+  b[1] = 7;
+  b[n - 2] = 7;
+  b[n - 1] = 6;
+
+  try {
+    CustomSelector selector(GetDeviceType(argc, argv));
+    queue q(selector, dpc_common::exception_handler);
+    cout << "[SYCL] Using device: ["
+         << q.get_device().get_info<info::device::name>() << "] from ["
+         << q.get_device().get_platform().get_info<info::platform::name>()
+         << "]\n";
+
+    // Create 1D buffers for b vector and iteration vectors.
+    buffer buffer_b(b, range{n});
+    buffer buffer_x_k(x_k, range{n});
+    buffer buffer_x_k1(x_k1, range{n});
+
+    for (int k = 0; k < 15000; k++) {
+      // k-th iteration of Jacobi.
+      q.submit([&](auto &h) {
+        auto acc_b = buffer_b.get_access<access::mode::read>(h);
```

Comment:
Should be using SYCL 2020 style accessors.

Reply:
Changed to `access_mode::read`.

Reply:
Now changed to accessor acc_b (buffer_b, h, read_only);